packages/project-access/src/library/helpers.ts

```diff
@@ -0,0 +1,291 @@
+    let reuseLib;
+    const manifestType = manifest['sap.app']?.type;
+    if ((manifestType === 'component' || manifestType === 'library') && manifestPath) {
+        const reuseType = getReuseType(manifestPath);
+        const libDeps = getManifestDependencies(manifest);
+        const description = getManifestDesc(manifest, manifestPath);
+
+        const libIndex = reuseLibs.findIndex((reuseLib) => reuseLib.name === manifest?.['sap.app'].id);
+        if (libIndex === -1) {
+            reuseLib = {
+                name: `${manifest['sap.app'].id}`,
+                path: manifestPath,
+                type: reuseType,
+                uri: manifest['sap.platform.abap']?.uri ?? '',
+                dependencies: libDeps,
+                libRoot: projectRoot,
+                description
+            };
+        }
+    }
+
+    return reuseLib;
+};
+
+/**
+ * Reads library file and returns a reuse lib object.
+ *
+ * @param library - library file content
+ * @param libraryPath - path to the library file
+ * @param projectRoot - root of the project
+ * @returns reuse library or undefined
+ */
+const getLibraryFromLibraryFile = async (
+    library: string,
+    libraryPath: string,
+    projectRoot: string
+): Promise<ReuseLib | undefined> => {
+    let libEntry;
+    const parsedFile = new XMLParser({ removeNSPrefix: true }).parse(library, false);
+    if (parsedFile?.library?.name) {
+        const manifestType = parsedFile?.library ? 'library' : 'component';
+        if (manifestType === 'component' || manifestType === 'library') {
+            const reuseType = getReuseType(libraryPath);
+            const libDeps = getLibraryDependencies(parsedFile);
+            const description = getLibraryDesc(parsedFile, libraryPath);
+            libEntry = {
+                name: `${parsedFile.library.name}`,
+                path: libraryPath,
+                type: reuseType,
+                uri: parsedFile.library?.appData?.manifest?.['sap.platform.abap']?.uri || '',
+                dependencies: libDeps,
+                libRoot: projectRoot,
+                description
+            };
+        }
+    }
+
+    return libEntry;
+};
+
+/**
+ * Updates the library options with the new library.
+ *
+ * @param reuseLibs - existing library options
+ * @param reuseLib - new library
+ */
+const updateLibOptions = (reuseLibs: ReuseLib[], reuseLib?: ReuseLib): void => {
+    if (reuseLib) {
+        const libIndex = reuseLibs.findIndex((lib) => lib.name === reuseLib.name);
+        if (libIndex >= 0) {
+            // replace
+            reuseLibs[libIndex] = reuseLib;
+        } else {
+            reuseLibs.push(reuseLib);
+        }
+    }
+};
+
+/**
+ * Returns an array of the reuse libraries found in the folders.
+ *
+ * @param libs - array of libraries found in the workspace folders.
+ * @returns list of reuse library
+ */
+export const getReuseLibs = async (libs?: LibraryResults[]): Promise<ReuseLib[]> => {
```

Comment:
Seems to overlap a lot with these functions https://github.com/SAP/open-ux-tools/blob/ee943db51305570cd3ecfe810d8a1010e1f10cc9/packages/project-access/src/project/search.ts#L369
Can the logic be shared?

Reply:
While there may be some overlap I'm not able to find a nice to consolidate any of the code.

I already utilise `findFioriArtifacts` (& sub function `filterLibraries`) earlier in the flow as it returns the root of the apps + libs in the workspace. 

`getReuseLibs` takes this result and goes deeper into each manifest/.library file and pulls out the relevant info for each lib or component so if it's much more involved.

Reply:
OK. Thanks for looking into it.